src/components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,143 @@
+			onCollapseChange( isCollapsed );
+		}
+	}, [ isCollapsed, onCollapseChange ] );
+
+	useEffect( () => {
+		const handleScreenResize = () => {
+			const isSmallScreen = window.innerWidth < 1280;
+			if ( isSmallScreen ) {
+				setIsCollapsed( true );
+				// eslint-disable-next-line no-undef
+				localStorage.setItem( 'sidebar-collapsed', JSON.stringify( true ) );
+			} else {
+				// eslint-disable-next-line no-undef
+				const storedState = localStorage.getItem( 'sidebar-collapsed' );
+				setIsCollapsed( storedState ? JSON.parse( storedState ) : false );
+			}
+
+			if ( sideBarRef.current ) {
+				if ( !! screenHeight ) {
+					sideBarRef.current.style.height = `${ window.innerHeight }px`;
+				} else {
+					sideBarRef.current.style.height = 'auto';
+				}
+			}
+		};
+
+		window.addEventListener( 'resize', handleScreenResize );
+		handleScreenResize();
+
+		return () => {
+			window.removeEventListener( 'resize', handleScreenResize );
+		};
+	}, [ screenHeight ] );
+
+	return (
+		<SidebarContext.Provider value={ { isCollapsed, setIsCollapsed, collapsible } }>
+			<div
+				ref={ sideBarRef }
+				className={ cn(
+					'overflow-auto w-72 px-4 py-4 gap-4 flex flex-col bg-background-primary',
+					borderOn &&
+						'border-0 border-r border-solid border-border-subtle',
+					!! screenHeight && 'h-screen',
+					'transition-all duration-200',
+					isCollapsed && 'w-16 px-2',
+					className
+				) }
+				{ ...props }
+			>
+				{ children }
+			</div>
+		</SidebarContext.Provider>
+	);
+};
+
+const SidebarHeader = ( { children } ) => {
+	return <div className="space-y-2">{ children }</div>;
+};
+
+const SidebarBody = ( { children } ) => {
+	return <div className={ cn( 'space-y-4 grow items-start' ) }>{ children }</div>;
+};
+
+const SidebarFooter = ( { children } ) => {
+	const { isCollapsed, setIsCollapsed, collapsible } =
+		useContext( SidebarContext );
+	return (
+		<div className="space-y-4">
+			{ children }
+			{ collapsible && (
+				<button
+					className={ cn(
+						'bg-transparent w-full border-0 p-0 m-0 flex items-center gap-2 text-base cursor-pointer',
+						isCollapsed && 'justify-center'
+					) }
+					onClick={ () => {
+						setIsCollapsed( ! isCollapsed );
+						// eslint-disable-next-line no-undef
+						localStorage.setItem(
+							'sidebar-collapsed',
+							JSON.stringify( ! isCollapsed )
+						);
+					} }
+					aria-label={
+						isCollapsed ? 'Expand sidebar' : 'Collapse sidebar'
+					}
+				>
+					{ isCollapsed ? (
+						<>
+							<Tooltip title="Expand" placement="right">
+								<PanelLeftOpen className="w-5 h-5" />
+							</Tooltip>
+						</>
+					) : (
+						<>
+							<PanelLeftClose className="w-5 h-5" /> Collapse
+						</>
+					) }
+				</button>
+			) }
+		</div>
+	);
+};
+
+const SidebarItem = ( { children, className } ) => {
+	return <div className={ cn( 'w-full', className ) }>{ children }</div>;
+};
+
+export { Sidebar, SidebarHeader, SidebarBody, SidebarFooter, SidebarItem };
```

Comment:
Can we follow the same syntax that we are following for other components? Ex.

```jsx
Sidebar.Item = SidebarItem

export Sidebar;
```

Reply:
As we discussed earlier it's causing issue when writing stories.
Code generated is different.

Reply:
If it's related to the component name then we can also handle that using `displayName`. Please let me know if it's not related to display name.

Reply:
IF we need to set it with display name. We are having to define. multiple stories.
So as other libraries also using in same way. We planned to change the names.

Reply:
We can define the name from the component so that we don't need to define it again. Please check the reference below: [Dialog Component](https://github.com/brainstormforce/force-ui/blob/4f24e75ec17cb0287f9958197dc579859939d31d/src/components/dialog/dialog.jsx#L159)

Reply:
For the parent it's adding the name but the children are still different. Are we able to set to children too?

Reply:
Yes. We can set it similarly. Please check the Dialog component for the reference.

Reply:
Okay, This would be good then for easy importing, Since we can set names in storybook I will change.